mnist-keras/client/data/read_data.py

```diff
@@ -1,13 +1,10 @@
 import numpy as np
-import os
 import tensorflow as tf
-from sklearn.model_selection import train_test_split
-from tensorflow import keras
 
-def read_data(path, nr_examples=1000,trainset=True):
+def read_data(path, trainset=True):
     """ Helper function to read and preprocess data for training with Keras. """
 
-    pack = np.load(path)
+    pack = np.load(path, allow_pickle=True)
```

Comment:
Why do we need allow_pickle=True? 

I think we should avoid pickling as much as possible.

Reply:
I got error from numpy that it could not read mnist.npz unless I set allow_pickle=True. I agree that this should be avoided but since I have no clue about how the original data (non-partitioned) was created this is want I did. If you can figure out how to avoid this please tell.

Reply:
Talk to Mattias. I think he has managed to avoid it in the past.

Reply:
I think I know where the problem is...

Reply:
fixed in [commit](https://github.com/scaleoutsystems/examples/pull/6/commits/80da1fdb9ab61bd2bb1f14cb0569a6798e1eb178) @ahellander @carmat88

Reply:
Great! Thank you for fixing this.